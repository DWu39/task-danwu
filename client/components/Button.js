import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import theme from '../style/theme';

const StyledButton = styled.button`
  background: ${({primary}) => primary ? theme.green : theme.gray};
  border: none;
  color: ${theme.white};
  cursor: pointer;
  padding: 8px;
  text-transform: capitalize;
`;

const Button = ({
  children,
  className,
  onClick,
  primary
}) => {
  return (
    <StyledButton
      className={className}
      onClick={onClick}
      primary={primary}
    >
      {children}
    </StyledButton>
  );
};

Button.propTypes = {
  children: PropTypes.any,
  className: PropTypes.any,
  onClick: PropTypes.func,
  primary: PropTypes.bool
}

Button.defaultProps = {
  children: null,
  className: null,
  onClick: null,
  primary: true
}

export default Button;
