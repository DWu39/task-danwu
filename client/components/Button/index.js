import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import theme from '../../style/theme';

const StyledButton = styled.button `
  background: ${ ({
  primary}) => primary ? theme.green : theme.gray};
  border: none;
  color: ${theme.white};
  cursor: pointer;
  padding: 12px;
  text-transform: capitalize;
`;

const Button = ({children, className, disabled, onClick, primary}) => {
  return (
    <StyledButton className={className} disabled={disabled} onClick={onClick} primary={primary}>
      {children}
    </StyledButton>
  );
};

Button.propTypes = {
  children: PropTypes.any,
  className: PropTypes.any,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  primary: PropTypes.bool
}

Button.defaultProps = {
  children: null,
  className: null,
  disabled: false,
  onClick: null,
  primary: true
}

export default Button;
