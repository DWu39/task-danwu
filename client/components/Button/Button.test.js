import React from 'react';
import { shallow } from 'enzyme';

import Button from './index';

describe('Button', () => {
  it('should render correctly', () => {
    const output = shallow(<Button>Press</Button>);
    expect(output).toMatchSnapshot();
  });
});
