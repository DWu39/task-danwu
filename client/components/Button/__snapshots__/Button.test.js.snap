// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className={null}
    disabled={false}
    onClick={null}
    primary={true}
  >
    Press
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Press",
      "className": null,
      "disabled": false,
      "onClick": null,
      "primary": true,
    },
    "ref": null,
    "rendered": "Press",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Press",
        "className": null,
        "disabled": false,
        "onClick": null,
        "primary": true,
      },
      "ref": null,
      "rendered": "Press",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
